src/iop/expfusion.c

```diff
@@ -0,0 +1,1268 @@
+  float exposure_width;      // exposure weight function variance
+  int weight_mode;           // algorithm used to build the weight map
+  int grey_projector;        // rgb --> grey
+  int fusion_colorspace;     // colorspace used to blend images
+  int fusion_grey_projector; // rgb --> grey if fusion_colorspace == rgb grey
+  float exposure_left_cutoff;
+  float exposure_right_cutoff;
+} dt_iop_fusion_params_t;
+
+typedef struct dt_iop_fusion_params_t dt_iop_fusion_data_t;
+
+typedef struct dt_iop_fusion_gui_data_t
+{
+  GtkWidget *sl_num_exposures;
+  GtkWidget *sl_exposure_stops;
+  GtkWidget *sl_exposure_optimum;
+  GtkWidget *sl_exposure_width;
+  GtkWidget *cmb_weight_mode;
+  GtkWidget *cmb_grey_projector;
+  GtkWidget *cmb_fusion_colorspace;
+  GtkWidget *cmb_fusion_grey_projector;
+  GtkWidget *sl_exposure_left_cutoff;
+  GtkWidget *sl_exposure_right_cutoff;
+} dt_iop_fusion_gui_data_t;
+
+const char *name()
+{
+  return _("exposure fusion");
+}
+
+int default_group()
+{
+  return IOP_GROUP_BASIC;
+}
+
+int flags()
+{
+  return IOP_FLAGS_SUPPORTS_BLENDING;
+}
+
+int default_colorspace(dt_iop_module_t *self, dt_dev_pixelpipe_t *pipe, dt_dev_pixelpipe_iop_t *piece)
+{
+  return iop_cs_rgb;
+}
+
+static void _show_hide_controls(dt_iop_fusion_params_t *p, dt_iop_fusion_gui_data_t *g)
+{
+  gtk_widget_set_visible(g->cmb_fusion_grey_projector, p->fusion_colorspace == DT_FUSION_COLORSPACE_RGB_GREY);
+}
+
+void commit_params(struct dt_iop_module_t *self, dt_iop_params_t *params, dt_dev_pixelpipe_t *pipe,
+                   dt_dev_pixelpipe_iop_t *piece)
+{
+  memcpy(piece->data, params, sizeof(dt_iop_fusion_params_t));
+}
+
+void init_pipe(struct dt_iop_module_t *self, dt_dev_pixelpipe_t *pipe, dt_dev_pixelpipe_iop_t *piece)
+{
+  piece->data = calloc(1, sizeof(dt_iop_fusion_data_t));
+  self->commit_params(self, self->default_params, pipe, piece);
+}
+
+void cleanup_pipe(struct dt_iop_module_t *self, dt_dev_pixelpipe_t *pipe, dt_dev_pixelpipe_iop_t *piece)
+{
+  free(piece->data);
+  piece->data = NULL;
+}
+
+void gui_update(struct dt_iop_module_t *self)
+{
+  dt_iop_fusion_params_t *p = (dt_iop_fusion_params_t *)self->params;
+  dt_iop_fusion_gui_data_t *g = (dt_iop_fusion_gui_data_t *)self->gui_data;
+
+  dt_bauhaus_slider_set(g->sl_num_exposures, p->num_exposures);
+  dt_bauhaus_slider_set(g->sl_exposure_stops, p->exposure_stops);
+  dt_bauhaus_slider_set(g->sl_exposure_optimum, p->exposure_optimum);
+  dt_bauhaus_slider_set(g->sl_exposure_width, p->exposure_width);
+  dt_bauhaus_combobox_set(g->cmb_weight_mode, p->weight_mode);
+  dt_bauhaus_combobox_set(g->cmb_grey_projector, p->grey_projector);
+  dt_bauhaus_combobox_set(g->cmb_fusion_colorspace, p->fusion_colorspace);
+  dt_bauhaus_combobox_set(g->cmb_fusion_grey_projector, p->fusion_grey_projector - 1);
+  dt_bauhaus_slider_set(g->sl_exposure_left_cutoff, p->exposure_left_cutoff * 100.f);
+  dt_bauhaus_slider_set(g->sl_exposure_right_cutoff, p->exposure_right_cutoff * 100.f);
```

Comment:
There should be some method of disabling the exposure cutoffs.  (Right now, 100% effectively disables it, but this will not continue to be the case if the algorithm is fixed to handle data >1.0f)